rts/Sim/Weapons/Weapon.cpp

```diff
@@ -1297,42 +1297,46 @@ float CWeapon::GetSafeInterceptTime(const CUnit* unit, float predictMult) const
 
 	// case 1, aa <0, target speed is less than projectile speed
 	// guaranteed existence of a positive solution
+	// case 1a, aa is a large value, standard quadratic formula works fine
 	if (aa < -1) {
 		// standard quadratic formula
 		predictTime = (-bb - math::sqrt(temp2 - temp1)) / (2 * aa);
 	}
+	// case 1b, aa is a small value, "inverted" standard quadratic formula works better, and extra check needed
 	else if (aa <= 0) {
 		// check catastrophic case of aa=0 and bb=0 
 		// target speed equal to projectile speed, and target is moving tangentally
-		if ((std::abs(aa) < (1.0 / (2 ^ 16))) && (std::abs(bb) < (1.0 / (2 ^ 16)))) {
-			return 0.0;
+		if ((std::abs(aa) < (std::pow(2, -16))) && (std::abs(bb) < (std::pow(2, -16)))) {
```

Comment:
consider making sure this is a compile-time constant, for example `constexpr auto VELOCITY_EPS = std::pow(2, -16);`